blockly-vue/src/components/BlocklyComponent.vue

```diff
@@ -29,30 +26,197 @@
 
 /**
  * @fileoverview Blockly Vue Component.
- * @author samelh@google.com (Sam El-Husseini)
+ * @author abbychau@gmail.com (Abby Chau)
  */
 
-import Blockly from 'blockly';
+import Blockly from "blockly";
+import Swal from "sweetalert2";
+
+var renameVar = function(name) {
+  return name;
+};
+
+Blockly.prompt = function(msg, defaultValue, callback) {
+  Swal.fire({
+    title: "Variable:",
+    input: "text",
+    inputAttributes: {
+      autocapitalize: "off"
+    },
+    showCancelButton: true,
+    confirmButtonText: "Look up",
+    showLoaderOnConfirm: true,
+    allowOutsideClick: () => !Swal.isLoading()
+  }).then(result => {
+    if (result.value) {
+      // console.log(result);
+      callback(renameVar(result.value));
+    }
+  });
+};
+
+Blockly.Blocks["stock_buy_simple"] = {
```

Comment:
Can we split this into another file under ``/src/blocks/``?

Reply:
see the react example as a reference: https://github.com/google/blockly-samples/blob/master/blockly-react/src/blocks/customblocks.js